sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -1106,16 +1105,16 @@ def extract_endpoints(stage):
                 key=key))
         controller.state_handler.blocking_append(state_key, elements_data)
 
-    def get_buffer(pcoll_id):
-      if (pcoll_id.startswith(b'materialize:')
-          or pcoll_id.startswith(b'timers:')):
-        if pcoll_id not in pcoll_buffers:
+    def get_buffer(buffer_id):
+      kind, name = split_buffer_id(buffer_id)
+      if kind in ('materialize, timers'):
+        if buffer_id not in pcoll_buffers:
           # Just store the data chunks for replay.
-          pcoll_buffers[pcoll_id] = list()
-      elif pcoll_id.startswith(b'group:'):
+          pcoll_buffers[buffer_id] = list()
```

Comment:
pcoll_buffers is keyed by whatever create_buffer_id returns; the fact that happens to be bytes isn't something the user should care about. (Well, it's bytes to stick it in a payload, but users of pcoll_buffers shouldn't care.)

I'll create a follow-up PR to make pcoll_buffers an ordinary dict which will be more robust, but don't have time and the moment and it would be good to unblock all these side input issues with Python 3.